great_expectations/expectations/expectation.py

```diff
@@ -764,7 +764,7 @@ def run_diagnostics(self, pretty_print=True):
                     expectation_config=expectation_config,
                 )
                 validation_result = validation_results[0]
-            except GreatExpectationsError as e:
+            except Exception as e:
                 report_obj = self._add_error_to_diagnostics_report(
                     report_obj, e, traceback.format_exc()
                 )
```

Comment:
@jcampbell thank you sending me down this rabbithole :) The appropriate error-checking already existed at `expectations/expectations.py` but the catch statement needed to be broader. 

Changing `GreatExpectationsError` to `Exception` does what you were asking for.

Reply:
as discussed in slack!